python/pyiceberg/io/pyarrow.py

```diff
@@ -470,6 +472,59 @@ def expression_to_pyarrow(expr: BooleanExpression) -> pc.Expression:
     return boolean_expression_visit(expr, _ConvertToArrowExpression())
 
 
+def open_task(
+    task: FileScanTask,
+    fs: FileSystem,
+    table: Table,
+    row_filter: BooleanExpression,
+    projected_schema: Schema,
+    case_sensitive: bool,
+) -> pa.Table:
+    _, path = PyArrowFileIO.parse_location(task.file.file_path)
+
+    bound_row_filter = bind(table.schema(), row_filter, case_sensitive=case_sensitive)
```

Comment:
I think we can bind this once to the table before fanning out. We can just pass in the `bound_row_filter` since this one will be the same for each of the tables (will also be easier on the GIL).

Reply:
This was to make `open_task` something that outside libraries could use. If we require a bound expression then I think that makes this harder for people to reuse. I'm fine with it as an internal method though.